internal/api/v1/testcases.go

```diff
@@ -237,3 +243,108 @@ func DeleteTestCase(testCaseService services.TestCaseService, logger logging.Log
 		})
 	}
 }
+
+// ImportTestCasesFromFile godoc
+//
+//	@ID				ImportTestCasesFromFile
+//	@Summary		Import test cases from Excel or CSV file
+//	@Description	Import test cases from Excel or CSV file
+//	@Tags			test-cases
+//	@Accept			json
+//	@Produce		json
+//	@Param			projectID formData	path		string	true	"Project ID"
+//	@Param			request	body		file	true	"Excel or CSV file"
+//	@Success		200			{object}	interface{}
+//	@Failure		400			{object}	problemdetail.ProblemDetail
+//	@Failure		500			{object}	problemdetail.ProblemDetail
+//	@Router			/v1/test-cases/import-file [post]
+func ImportTestCasesFromFile(testCaseService services.TestCaseService, logger logging.Logger) fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		projectIDStr := c.FormValue("projectID")
+		projectID, err := strconv.ParseInt(projectIDStr, 10, 64)
+		if err != nil || projectID == 0 {
+			return problemdetail.BadRequest(c, "invalid project ID")
+		}
+
+		fileHeader, err := c.FormFile("file")
+		if err != nil {
+			return problemdetail.BadRequest(c, "file is required")
+		}
+
+		file, err := fileHeader.Open()
+		if err != nil {
+			return problemdetail.ServerErrorProblem(c, "failed to open uploaded file")
+		}
+		defer file.Close()
+
+		ext := strings.ToLower(filepath.Ext(fileHeader.Filename))
+		var rows [][]string
+
+		if ext == ".csv" {
+			reader := csv.NewReader(file)
+			rows, err = reader.ReadAll()
+			if err != nil {
+				return problemdetail.BadRequest(c, "failed to parse CSV file")
+			}
```

Comment:
this shouldn't be a bad request, the failure to read the file is on the server side - also log the error